Sources/SpeziFoundation/Misc/TimeoutError.swift

```diff
@@ -0,0 +1,72 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+
+
+/// Timeout occurred inside an async operation.
+public struct TimeoutError {
+    /// Create a new timeout error.
+    public init() {}
+}
+
+
+extension TimeoutError: Error {}
+
+
+/// Race a timeout.
+///
+/// This method can be used to race an operation against a timeout.
+///
+/// ### Timeout in Async Context
+///
+/// Below is a code example showing how to best use the `withTimeout(of:perform:)` method in an async method.
+/// The example uses [Structured Concurrency](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/concurrency)
+/// creating a child task running the timeout task. This makes sure that the timeout is automatically cancelled when the method goes out of scope.
+///
+/// ```swift
+/// func foo() async throws {
+///     async let _ = withTimeout(of: .seconds(30)) {
+///         // cancel `operation` method below (e.g., resume continuation by throwing a `TimeoutError`)
+///     }
+///
+///     try await operation()
+/// }
```

Comment:
I think it would make sense to add a more complete example here, e.g., with a continuation included. In the current setup I don't see a nice way someone would understand how to cancel the operation and I wouldn't be aware of a different mechanism than a continuation?

Reply:
Make sense. I initially added a continuation-based example and decided to simplify it for more clarity. But seems like that a more detailed code example might actually be worth it to help demonstrate a basic use case for this method.